src/main/java/io/datanerds/avropatch/Patch.java

```diff
@@ -35,16 +45,29 @@ public boolean addOperation(Operation operation) {
         return Collections.unmodifiableList(operations);
     }
 
+    public Map<String, ?> getHeaders() {
```

Comment:
Agreed that it is not open to modifications to external modifiers but from a code extensibility perspective, if there is a method in this class which tries to modify this map, extra care would need to be taken to ensure that the map is made immutable then. By making it immutable now, we can future proof it so that we don't run the risk of somebody not making it immutable/thread safe while adding code modifying it in future.